packages/openactive-broker-microservice/src/util/item-transforms.js

```diff
@@ -0,0 +1,91 @@
+const { partition, omit } = require('lodash');
+
+/**
+ * Inverts any items that have an `individualFacilityUse` property, so that the top-level `kind` is "IndividualFacilityUse"
+ *
+ * @param {{state: string, modified:any, kind: string, id: string, data: {individualFacilityUse: object}}[]} items
+ */
+function invertFacilityUseItems(items) {
+  const [invertibleFacilityUseItems, otherItems] = partition(items, (item) => item.data?.individualFacilityUse?.length > 0);
+  if (invertibleFacilityUseItems.length < 1) return items;
+
+  // Invert "FacilityUse" items so the the top-level `kind` is "IndividualFacilityUse"
+  const invertedItems = [];
+  for (const facilityUseItem of invertibleFacilityUseItems) {
+    for (const individualFacilityUse of facilityUseItem.data.individualFacilityUse) {
+      invertedItems.push({
+        ...facilityUseItem,
+        kind: individualFacilityUse['@type'],
+        id: individualFacilityUse['@id'],
+        data: {
+          ...individualFacilityUse,
+          '@context': facilityUseItem.data['@context'],
+          aggregateFacilityUse: omit(facilityUseItem.data, ['individualFacilityUse', '@context']),
+        },
+      });
+    }
+  }
+
+  return invertedItems.concat(otherItems);
+}
+
+/**
+ * Creates an opportunity item from a subEvent. This is used when a SessionSeries feed has ScheduledSessions,
+ * and using this function splits the subEvents into items to simulate a ScheduledSessions feed.
+ *
+ * @param {object} subEvent
+ * @param {{modified: any, data: object}} item
+ */
+function createItemFromSubEvent(subEvent, item) {
+  const opportunityItemData = {
+    ...subEvent,
+  };
+  opportunityItemData['@context'] = item.data['@context'];
+  opportunityItemData.superEvent = item.data['@id'] || item.data.id;
+
+  const opportunityItem = {
+    id: subEvent['@id'] || subEvent.id,
+    modified: item.modified,
+    kind: subEvent['@type'] || subEvent.type,
+    state: 'updated',
+    data: opportunityItemData,
+  };
+
+  return opportunityItem;
+}
+
+/**
+ * @param {{superEvent?: any, facilityUse?: any}} data
+ */
+function jsonLdHasReferencedParent(data) {
+  return typeof data?.superEvent === 'string' || typeof data?.facilityUse === 'string';
+}
+
+/**
+ * Sorts the JSON-LD context array, with https://openactive.io/ and https://schema.org/ at the top.
+ *
+ * @param {string[]} arr
+ */
+function sortWithOpenActiveOnTop(arr) {
+  const firstList = [];
+  if (arr.includes('https://openactive.io/')) firstList.push('https://openactive.io/');
+  if (arr.includes('https://schema.org/')) firstList.push('https://schema.org/');
+  const remainingList = arr.filter((x) => x !== 'https://openactive.io/' && x !== 'https://schema.org/');
+  return firstList.concat(remainingList.sort());
+}
+
+/**
+ * Merges and sorts contexts
+ *
+ * @param  {object[]} opportunities
```

Comment:
`object` is weird in TypeScript. e.g. it will accept arrays and other kinds of objects. `Record<string, any>` is a bit more useful for these sorts of things